### DISCUSSION TITLE
CWatM coupled with MODFLOW

### DISCUSSION THREAD

**Discussion by Zhyu-1** [iiasa/CWatM#15] [Q&A] [answered]
Hi,

In the model's manual, CWatM can integrate with MODFLOW, but there are no examples for users how to coupled with MODFLOW. Can you provide some examples about how to integrated CWatM and MODFLOW?

Thanks!
Jade

**Comment by guillaumot** [comment]
Hi Jade,
Thanks for your question. We will add the CWatM-MODFLOW in our repository very soon.
Best,
Luca

**Reply by Zhyu-1** [reply]
Hi,
Thanks for your updated CWatM version. I have studied the coupled CWatM-Modflow model several times, but I still could not run the model successfully. Could you please upload an example which includes "settings.ini"(include MODFLOW options) and the input parameters(MODFLOW model needs)? Thanks for your answer.
Best,
Jade

**Reply by hsk0059** [reply]
Hi Jade,

I am also studying CWatM-Modflow now, could we discuss offline if you are still studying it?

**Reply by weixiaoxun** [reply]
@hsk0059 您好，何师兄，我也在学习CWatM-Modflow，如果您还在学习的话，可以请教您一些问题吗

**Reply by hsk0059** [reply]
> @hsk0059 您好，何师兄，我也在学习CWatM-Modflow，如果您还在学习的话，可以请教您一些问题吗

@weixiaoxun xiaoxun您好，抱歉才看到消息；我也是刚入门，可以互相交流，向你学习；我邮箱he_shaokun@whu.edu.cn；之后可以微信交流！

**Comment by guillaumot** [comment]
Dear Jade,
happy that you are interested by the coupled CWatM-ModFlow version!
Sorry we updated the Github repository but we did not add the pre-processing part to set up the ModFlow model (with a settings file, Python codes to create input files and readme/requirement files). We keep you informed soon.
Best,
Luca

**Comment by guillaumot** [comment, marked as answer]
[cwatmmodflow6_github.zip](https://github.com/iiasa/CWatM/files/7569445/cwatmmodflow6_github.zip)

Hi Jade,
please find attached the CWATM-MODFLOW6  version. First, the model require to create the Modflow grid. So , you need to run _Main_Preprocess_ModFlow.py_ using your CWATM mask map and one topographic map. Inside this code, you need to define the resolution you wish and the coordinates system for ModFlow and CWatM, the MODFLOW coordinates system as to be regular (square grid).

Inside 'cwatm' folder, there is one settings file given as example.

Let me know if you have questions.
Luca

**Reply by Zprivf** [reply]
Hello @guillaumot, thank you for sharing CWatM - Modflow coupling example. 
My name is Fitsume and I am a Ph.D. student at the University of Alabama. I am using CWatM for a global water scarcity study. So far I have managed to successfully run the model. I am currently trying to couple the model with MODFLOW. Using the sample file provided in here, I have prepared the Modflow pre-processing files and run the coupled CWatM model with it (it's a global model). However, after running the CWatM-Modflow Model I couldn’t get the CWatM outputs. The result only shows the MODFLOW outputs. Do I have to set additional parameters in the setting file to get the CWatM output (outputs like discharge, runoff...). 
 I have edited the setting file you shared in this thread, which is attached below.

Thank you so much for the help.

[settings_Global_forModflow.txt](https://github.com/iiasa/CWatM/files/10828507/settings_Global_forModflow.txt)

**Reply by guillaumot** [reply]
Hi Fitsume,
Happy that you manage to set up everything. Quite impressive that you did it at global scale.
I had a look at your settings:
All variables in the code written "self.var.name-of-the-variable" can be added to the outputs. You need to add them (without the "self.var.") at the end of the settings file.
For example, this line:
OUT_Map_MonthAvg = modflow_watertable, sum_gwRecharge, capillar, discharge, tws, totalSto, lakeResStorage, channelStorage, sum_soil, sum_interceptStor, sum_topwater, sum_w1, sum_w2, sum_w3, SnowCover, groundwater_storage_available, totalIrrDemand, modfPumpingM, sum_runoff, baseflow
will give you all these variables at a monthly time step, the spatial resolution is the CWatM resolution unless for variables starting with "modflow" like "modflow_watertable" in this example.

Just in case, I saw in the settings file that "modflow_coupling = False" and that water demand is True while Groundwater_pumping = False.
So, I think that modlfow coupling is not really activated here and I am not sure that you really want to activate demand without groundwater pumping.

Best regards,
Luca

**Reply by Zprivf** [reply]
Thank you, Luca, this is very helpful. I apologize I sent the files that were not activated.
